lib/core/include/metadata.hpp

```diff
@@ -0,0 +1,232 @@
+#ifndef METADATA_HPP
+#define METADATA_HPP
+
+#include <string>
+#include <vector>
+#include <unordered_map>
+
+#include "entity.hpp"
+#include "irods_query.hpp"
+
+#ifdef RODS_SERVER
+    #include "rsModAVUMetadata.hpp"
+    using rxComm_t = rsComm_t;
+    #define rxModAVUMetadata rsModAVUMetadata
+    #define IRODS_FILESYSTEM_ENABLE_SERVER_SIDE_API
+#else
+    #include "modAVUMetadata.h"
+    using rxComm_t = rcComm_t;
+    #define rxModAVUMetadata rcModAVUMetadata
+#endif
+
+#include "filesystem.hpp"
+#include "fmt/format.h"
+
+namespace irods::experimental::metadata {
+
+    struct avu {
+        std::string attribute;
+        std::string value;
+        std::string units;
+    };
+
+    auto operator==(const avu& _lhs, const avu& _rhs) {
+        bool eq = _lhs.attribute == _rhs.attribute &&
+                  _lhs.value     == _rhs.value &&
+                  _lhs.units     == _rhs.units;
+        return eq;
+    }
+
+    namespace tokens {
+        const std::string none{};
+        const std::string collection{"-C"};
+        const std::string object{"-d"};
+        const std::string user{"-u"};
+        const std::string resource{"-R"};
+    }
```

Comment:
is this an implementation detail?  if so, this can be moved into the anonymous namespace below.